### PR TITLE
A solution for the first homework (hwdetect)

### PR DESCRIPTION
**Pull request by Renalime**
A bash script that detects whether sd cards, flash drives, i2c devices and usb-ttl convertors are connected. 
Each hardware type is served by a separate function with a descriptive name.
All the caveats are described in commit messages.

### CODE REVIEW

File: hwdetect/hwdetect.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+check_usb_to_ttl() {
+	usb_to_ttl=$(ls /dev | grep -E "^ttyUSB[0-9]+$")
+	if [ -n "$usb_to_ttl" ];
+	then
+		echo "A USB to TTL convertor is connected."
```
Comment (yekovalyov) [inline hwdetect/hwdetect.sh]: Please add device file name into the message.

File: hwdetect/hwdetect.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		then
+			echo "A USB flash drive is connected."
+		fi
+	done
+}
+
+check_sd_card() {
+	if [ -n "$(ls /dev | grep -E "^mmcblk[0-9]+$")" ];
+	then
+		echo "An SD card is connected."
```
Comment (yekovalyov) [inline hwdetect/hwdetect.sh]: Please add device file name into the message.

File: hwdetect/hwdetect.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	fi
+}
+
+check_i2c_devices() {
+	for ((i=1; i<=$(i2cdetect -l | wc -l); i++));
+	do
+		i2c_bus=$(i2cdetect -y ${i} | awk '{if (NR!=1) {print}}' | awk '{$1=""}1')
+		if [ -n "$(echo $i2c_bus | grep '[0-9]')" ];
+		then
+			echo "An i2c device is connected"
```
Comment (yekovalyov) [inline hwdetect/hwdetect.sh]: Please add i2c bus number.

**Comment by Renalime** [comment]
@yekovalyov Done all the requested features, although it doesn't look pretty, since sd card and usb to ttl functions basically spits out all the devices that were found by grep. But it works.

**Comment by Renalime** [comment]
@yekovalyov One more thing, should I remove 'question' label after the requested features are ready?

**Pull request merged**